### PR TITLE
feat(msg-parser/gate): Implemented `parse_candlestick()` for gate

### PR DESCRIPTION
**Pull request by iyangdianfeng**
Implemented `parse_candlestick()` for gate

### CODE REVIEW

File: crypto-msg-parser/src/exchanges/gate/gate_spot_current.rs [hunk 1]
```diff
@@ -309,3 +309,71 @@ pub(super) fn parse_bbo(msg: &str) -> Result<Vec<BboMsg>, SimpleError> {
...
+        ))
+    })?;
+
+    let ch: Vec<&str> = obj.result.n.split("_").collect();
+    let symbol = [ch[1].to_string(), ch[2].to_string()].join("_");
+
+    // m, minute; H, hour; D, day; W, week; M, month; Y, year
+    // interval	10s
+
+    // let period = ch[0].to_string();
```
Comment (pan93412) [inline crypto-msg-parser/src/exchanges/gate/gate_spot_current.rs:342]: Better to remove this dead code.

File: crypto-msg-parser/src/exchanges/gate/gate_swap.rs [hunk 2]
```diff
@@ -562,3 +565,77 @@ pub(super) fn parse_bbo(market_type: MarketType, msg: &str) -> Result<Vec<BboMsg
...
+#[derive(Serialize, Deserialize)]
+struct RawKlineMsg {
+    t: i64,    // 秒 s 精度的 Unix 时间戳
+    v: f64,    // integer交易量，只有市场行情的 K 线数据里有该值
+    c: String, // 收盘价
+    h: String, // 最高价
+    l: String, // 最低价
+    o: String, // 开盘价
+    n: String,
+    a: String, //基础货币交易量
```
Comment (pan93412) [inline crypto-msg-parser/src/exchanges/gate/gate_swap.rs:578]: Add space between `//` and `基礎`.

File: crypto-msg-parser/src/exchanges/gate/gate_swap.rs [hunk 3]
```diff
@@ -562,3 +565,77 @@ pub(super) fn parse_bbo(market_type: MarketType, msg: &str) -> Result<Vec<BboMsg
...
+    let mut kline_msgs = Vec::with_capacity(obj.result.len());
+
+    for kline in obj.result.iter() {
+        let ch: Vec<&str> = kline.n.split("_").collect();
+        let symbol = [ch[1].to_string(), ch[2].to_string()].join("_");
+
+        // m, minute; H, hour; D, day; W, week; M, month; Y, year
+        // interval	10s
+
+        // let period = ch[0].to_string();
```
Comment (pan93412) [inline crypto-msg-parser/src/exchanges/gate/gate_swap.rs:601]: Better to remove this dead code.

**Review by pan93412 — CHANGES_REQUESTED** [review]

**Comment by pan93412** [comment]
Besides, you can run `cargo fmt` to format your code.

**Review by pan93412 — APPROVED** [review]

**Review by soulmachine — CHANGES_REQUESTED** [review]
@iyangdianfeng don't use Chinese in comment, using English is a good habit, which can bring your project to wider audience over the world

**Review by soulmachine — APPROVED** [review]

**Pull request merged**